app/controllers/api/vms_controller.rb

```diff
@@ -53,6 +54,98 @@ def edit_resource(type, id, data)
     def delete_resource(type, id, _data = nil)
       enqueue_ems_action(type, id, "Deleting", :method_name => "destroy")
     end
+ 
+    def vm_reconfigure_resource(type, id, data)
+    
+      new_row = {
+        :id        => id.to_s,
+        :long_id   => id.to_s,
+        :cells     => []
+      }
+      
+      unless id
+          data_spec = data.collect { |key, val| "#{key}=#{val}" }.join(", ")
+          raise NotFoundError, "Invalid #{type} resource specified - #{data_spec}"
+        end
```

Comment:
Prefer a guard clause right at the beginning... Do we need to say what id was invalid?
Fixed indents.

```suggestion
      unless id
        data_spec = data.collect { |key, val| "#{key}=#{val}" }.join(", ")
        raise NotFoundError, "Invalid #{type} resource specified - #{data_spec}"
      end

      new_row = {
        :id        => id.to_s,
        :long_id   => id.to_s,
        :cells     => []
      }
```